test/e2e/tests/xcall.go

```diff
@@ -4,64 +4,104 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"github.com/icon-project/ibc-integration/test/testsuite"
-	"strings"
-	"testing"
-
 	"github.com/icon-project/ibc-integration/test/chains"
+	"github.com/icon-project/ibc-integration/test/testsuite"
 	"github.com/strangelove-ventures/interchaintest/v7/ibc"
 	"github.com/stretchr/testify/assert"
+	"strconv"
+	"strings"
+	"testing"
 )
 
 type XCallTestSuite struct {
 	*testsuite.E2ETestSuite
 	T *testing.T
 }
 
-func (x *XCallTestSuite) TestDemo() {
+func (x *XCallTestSuite) TextXCall() {
+	testcase := "xcall"
 	portId := "transfer"
-	ctx := context.TODO()
-	x.SetupXCall(ctx, portId)
-	x.DeployXCallMockApp(ctx, portId)
-
+	ctx := context.WithValue(context.TODO(), "testcase", testcase)
+	x.Require().NoError(x.SetupXCall(ctx, portId, 100), "fail to setup xcall")
+	x.Require().NoError(x.DeployXCallMockApp(ctx, portId), "fail to deploy xcall dapp")
+	chainA, chainB := x.GetChains()
 	x.T.Run("xcall one way message chainA-chainB", func(t *testing.T) {
-		chainA, chainB := x.GetChains()
-		x.TestOneWayMessage(ctx, t, chainA, chainB)
+
+		x.testOneWayMessage(ctx, t, chainA, chainB)
 	})
 
 	x.T.Run("xcall one way message chainB-chainA", func(t *testing.T) {
-		chainA, chainB := x.GetChains()
-		x.TestOneWayMessage(ctx, t, chainB, chainA)
+
+		x.testOneWayMessage(ctx, t, chainB, chainA)
 	})
 
 	x.T.Run("xcall test rollback chainA-chainB", func(t *testing.T) {
-		chainA, chainB := x.GetChains()
-		x.TestRollback(ctx, t, chainA, chainB)
+		x.testRollback(ctx, t, chainA, chainB)
 	})
 
 	x.T.Run("xcall test rollback chainB-chainA", func(t *testing.T) {
-		chainA, chainB := x.GetChains()
-		x.TestRollback(ctx, t, chainB, chainA)
+		x.testRollback(ctx, t, chainB, chainA)
 	})
 
 	x.T.Run("xcall test send maxSize Data: 2048 bytes", func(t *testing.T) {
-		chainA, chainB := x.GetChains()
-		x.TestOneWayMessageWithSize(ctx, t, 1300, chainA, chainB)
-		x.TestOneWayMessageWithSize(ctx, t, 1868, chainB, chainA)
+		x.testOneWayMessageWithSize(ctx, t, 1300, chainA, chainB)
+		x.testOneWayMessageWithSize(ctx, t, 1300, chainB, chainA)
 	})
 
 	x.T.Run("xcall test send maxSize Data: 2049bytes", func(t *testing.T) {
-		chainA, chainB := x.GetChains()
-		x.TestOneWayMessageWithSizeExpectingError(ctx, t, 1869, chainB, chainA)
-		x.TestOneWayMessageWithSizeExpectingError(ctx, t, 2000, chainA, chainB)
+		x.testOneWayMessageWithSizeExpectingError(ctx, t, 2000, chainB, chainA)
+		x.testOneWayMessageWithSizeExpectingError(ctx, t, 2000, chainA, chainB)
 	})
 
 }
 
-func (x *XCallTestSuite) TestOneWayMessage(ctx context.Context, t *testing.T, chainA, chainB chains.Chain) {
+func (x *XCallTestSuite) TestXCallPacketDrop() {
```

Comment:
Wouldn't this test be better to be done with a two way message and rollback via timeout error. Testing the whole timeout flow?

Reply:
I have revised the test flow and pushed the changes.